src/transforms/eigenanalysis.jl

```diff
@@ -104,75 +112,98 @@ function reapply(transform::EigenAnalysis, table, cache)
     assertion(table)
   end
 
-  # original columns names
-  cols = Tables.columns(table)
-  names = Tables.columnnames(cols)
-
   # table as matrix
   X = Tables.matrix(table)
 
+  # output dim
+  ndim = _ndim(transform.ndim, X)
+
   # retrieve cache
-  μ, S, S⁻¹ = cache
+  μ, S, S⁻¹, onames = cache
 
   # center the data
   Y = X .- μ
 
   # project the data
   Z = Y * S
 
+  # column names
+  names = Symbol.(:PC, 1:ndim)
+
   # table with transformed columns
   𝒯 = (; zip(names, eachcol(Z))...)
   𝒯 |> Tables.materializer(table)
 end
 
-function eigenmatrices(transform, Y)
+function eigenmatrices(transform, Y, ndim)
   proj = transform.proj
 
   Σ = cov(Y)
   λ, V = eigen(Σ)
 
   if proj == :V
-    S   = V
-    S⁻¹ = transpose(V)
+    P = V
   elseif proj == :VD
-    Λ   = Diagonal(sqrt.(λ))
-    S   = V * inv(Λ)
-    S⁻¹ = Λ * transpose(V)
+    Λ = Diagonal(sqrt.(λ))
+    P = V * inv(Λ)
   elseif proj == :VDV
-    Λ   = Diagonal(sqrt.(λ))
-    S   = V * inv(Λ) * transpose(V)
-    S⁻¹ = V * Λ * transpose(V)
+    Λ = Diagonal(sqrt.(λ))
+    P = V * inv(Λ) * transpose(V)
   end
 
+  Pp  = P[:, end:-1:1]
```

Comment:
Instead of reversing the matrix here, try to use the sortby option in eigen to order the eigenvalues in descending order.

Reply:
If eigenvalues ​​and eigenvectors are reversed, the output will be slightly different, breaking compatibility with the current reference test.

Reply:
I think that is fine. We just need to review the results carefully and add this change as a minor release.

Reply:
I didn't know we had tests that depended on the order of the vectors in the projection BTW.

Reply:
It's not just a change in the x and y order, it's a change in the output values.

Reply:
What change exactly? Can you please clarify what tests fail if we flip the order of vectors?

Reply:
The S matrix will be slightly different because `inv(Λ)` will return a slightly different matrix.

Reply:
That's why I had to change the reference tests in the previous PR.

Reply:
> The `S` matrix will be slightly different because `inv(Λ)` will return a slightly different matrix.

That was not my question, but the `inv(Λ)` expression before and after the proposed change should be the same except for the order of the entries in the diagonal. This is a diagonal matrix.

Can you please share which line in the test suite fails when you just flip the order without discarding vectors from the basis?

Reply:
In visual tests, the plots will be different if I change the order of the eigenvalues.